Fixed base/locks.jl documentation (Issue #44897)

GHTaarn opened the pull request:
My proposal for fixing Issue #44897

GHTaarn committed:
Fixed base/locks.jl documentation (Issue #44897)

fredrikekre commented on base/lock.jl:
```diff
@@ -221,7 +221,7 @@ When this function returns, the `lock` has been released, so the caller should
 not attempt to `unlock` it.
 
 !!! compat "Julia 1.7"
-    Using a [`Channel`](@ref) as the second argument requires Julia 1.7 or later.
```
Looks like this is actually correct, see https://github.com/JuliaLang/julia/commit/d9f0f04477bde51d858400a123cf40ccf4adc38d. Perhaps the best would be to update this docstring to a generic two-argument `lock` method?

GHTaarn reviewed (commented):
You are right about Channel being a possible second argument, but there is still the issue that using any two arguments is a feature that was only introduced in 1.7. Prior to that lock only took one argument. I will make a new proposal.

Do you know exactly what Julia does when the second argument is a Channel? It would be good to get this detail into the docstring as well.

GHTaarn committed:
Improved docstring for 2 argument lock method

Mentioning Channel is reintroduced.

It would be good to increase the detail for Channel as second argument

GHTaarn commented:
Yes, you are right about the two argument version being in earlier versions. In that case, disregard my latest suggestion, yours is better.
